### PR TITLE
add pass through macro

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**What change does this PR introduce?** 

The addition of the `add_pass_through_columns` macro. This macro allows a user to declare a pass through variable in the following format:
```
dbt_project.yml

...

vars:

  deal_pass_through_columns:
  
        - name: "property_sales_development_representative"
          alias: "sales_development_representative_owner_id"
  
        - name: "property_energy_consultant"
          alias: "energy_consultant_owner_id"

```

**If this PR introduces a new macro, how did you test the new macro?** 

This macro was tested within the dbt_hubspot_source package and was originally created by @jamesrayoub within the respective [PR](https://github.com/fivetran/dbt_hubspot_source/pull/35)

**If this PR introduces a modification to an existing macro, which packages is the macro currently present in and what steps were taken to test compatibility across packages?** 

This will only be used within the dbt_hubspot_source package. However, it can be applied to other packages that accept pass through fields as variables.

**Did you update the README to reflect the macro addition/modifications?** 

- [x] Yes
- [ ] No (provide further explanation)

### CODE REVIEW

**Comment by fivetran-joemarkiewicz** [comment]
An additional macro was added to allow for the code within the staging model to remain dry and dynamic. The `fill_pass_through_columns` macro was tested and documented.

File: macros/add_pass_through_columns.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% macro add_pass_through_columns(base_columns, pass_through_var) %}
+
+  {% if pass_through_var %}
+
+    {% for column in pass_through_var %}
+
+      {% do base_columns.append({ "name": column.name, "datatype": column.datatype, "alias": column.alias }) if column.alias else base_columns.append({ "name": column.name, "datatype": column.datatype}) %}
```
Comment (fivetran-jamie) [inline macros/add_pass_through_columns.sql:7]: are we removing the datatype attribute from this part? forgot where we left off

Comment (fivetran-joemarkiewicz) [reply macros/add_pass_through_columns.sql]: Yeah I think it makes the most sense to remove it as it is not needed and adds unnecessary complexity. Removed, thanks!

**Review by fivetran-jamie — APPROVED** [review]
looks good! i would just maybe add some more documentation for the macros (like in what types of files they're called in) but the code is all good

**Pull request merged**